lib/pavilion/series.py

```diff
@@ -36,12 +51,322 @@ def test_obj_from_id(pav_cfg, test_ids):
+                mb_tracker=mb_tracker,
+                build_only=False,
+                rebuild=False
+            )
+
+        except commands.CommandError as err:
+            # probably won't happen
+            err = codecs.decode(str(err), 'unicode-escape')
+            self.finished.append(test_name)
+            return None
+
+        except test_config.file_format.TestConfigError as err:
+            self.finished.append(test_name)
+            return None
+
+        if tests_by_sched is None:
+            # probably won't happen but just in case
+            self.finished.append(test_name)
+            return None
+
+        all_tests = sum(tests_by_sched.values(), [])
+        run_cmd.last_tests = list(all_tests)
+
+        if not all_tests:
+            # probably won't happen but just in case
+            self.test_info[test_name]['obj'] = run_cmd.last_tests
+            self.finished.append(test_name)
+            return None
+
+        # assign test to series and vice versa
+        self.series_obj.add_tests(all_tests)
+        run_cmd.last_series = self.series_obj
+        self.test_info[test_name]['obj'] = run_cmd.last_tests
+
+        # make sure result parsers are ok
+        res = run_cmd.check_result_format(all_tests)
+        if res != 0:
+            run_cmd.complete_tests(all_tests)
+            self.finished.append(test_name)
+            return None
+
+        # attempt to build
+        res = run_cmd.build_local(
+            tests=all_tests,
+            max_threads=self.pav_cfg.build_threads,
+            mb_tracker=mb_tracker,
+            build_verbosity=0
+        )
+        if res != 0:
+            run_cmd.complete_tests(all_tests)
+            self.finished.append(test_name)
+            return None
+
+        if self.series_cfg['simultaneous'] is None:
+            run_cmd.run_tests(
+                pav_cfg=self.pav_cfg,
+                tests_by_sched=tests_by_sched,
+                series=self.series_obj,
+                wait=None,
+                report_status=False
+            )
+            self.started.append(test_name)
+        else:
+            simult = int(self.series_cfg['simultaneous'])
+
+            # [ { sched: test_obj}, { sched: test_obj }, etc. ]
+            list_of_tests_by_sched = []
+            for sched, test_objs in tests_by_sched.items():
+                for test_obj in test_objs:
+                    list_of_tests_by_sched.append({sched: [test_obj]})
+
+            self.started.append(test_name)
+            for test in list_of_tests_by_sched:
+                self.test_wait(simult)
+                run_cmd.run_tests(
+                    pav_cfg=self.pav_cfg,
+                    tests_by_sched=test,
+                    series=self.series_obj,
+                    wait=None,
+                    report_status=False
+                )
+
+    def test_wait(self, simul):
+
+        while len(self.get_currently_running()) >= simul:
+            time.sleep(5)
```

Comment:
Don't sleep this long. Sleep an amount of time that is unnoticeable to a human, but an eternity to a computer. 
I find 0.01 or 0.1 seconds hits that spot.